packages/edit-site/src/components/sidebar-navigation-screen-page/index.js

```diff
@@ -18,42 +35,235 @@ import SidebarNavigationScreen from '../sidebar-navigation-screen';
+					: __( 'Untitled' );
+			} else {
+				_parentTitle = __( 'Top level' );
+			}
+
+			return {
+				parentTitle: _parentTitle,
+				templateSlug: getEditedPostContext()?.templateSlug,
+				featuredMediaDetails: {
+					...getFeaturedMediaDetails( attachedMedia, postId ),
+					altText: escapeAttribute(
+						attachedMedia?.alt_text ||
+							attachedMedia?.description?.raw ||
+							''
+					),
+				},
+			};
+		},
+		[ record, postId ]
+	);
+
+	// Match template slug to template title.
+	const { records: templates, isResolving: areTemplatesLoading } =
+		useEntityRecords( 'postType', 'wp_template', {
+			per_page: -1,
+		} );
+	const templateTitle =
+		! areTemplatesLoading && templateSlug
+			? templates?.find( ( template ) => template?.slug === templateSlug )
+					?.title?.rendered || templateSlug
+			: '';
+
+	return record ? (
 		<SidebarNavigationScreen
-			title={ record ? decodeEntities( record?.title?.rendered ) : null }
+			title={ decodeEntities( record?.title?.rendered ) }
 			actions={
 				<SidebarButton
 					onClick={ () => setCanvasMode( 'edit' ) }
 					label={ __( 'Edit' ) }
 					icon={ pencil }
 				/>
 			}
-			description={ __(
-				'Pages are static and are not listed by date. Pages do not use tags or categories.'
-			) }
+			meta={
+				<ExternalLink
+					className="edit-site-sidebar-navigation-screen__page-link"
+					href={ record.link }
+				>
+					{ record.link.replace( /^(https?:\/\/)?/, '' ) }
+				</ExternalLink>
+			}
 			content={
 				<>
-					{ record?.link ? (
-						<ExternalLink
-							className="edit-site-sidebar-navigation-screen__page-link"
-							href={ record.link }
+					<VStack
+						className="edit-site-sidebar-navigation-screen-page__featured-image-wrapper"
+						alignment="left"
+						spacing={ 2 }
+					>
+						<div
+							className={ classnames(
+								'edit-site-sidebar-navigation-screen-page__featured-image',
+								{
+									'has-image': !! mediaSourceUrl,
+								}
+							) }
+						>
+							{ !! mediaSourceUrl && (
+								<img
+									alt={
+										altText
+											? decodeEntities( altText )
+											: decodeEntities(
+													record.title?.rendered
+											  ) || __( 'Featured image' )
+									}
+									src={ mediaSourceUrl }
+								/>
+							) }
+							{ ! record?.featured_media && (
+								<p>{ __( 'No featured image' ) }</p>
+							) }
+						</div>
+					</VStack>
+					{ !! record?.excerpt?.rendered && (
+						<Truncate
+							className="edit-site-sidebar-navigation-screen-page__excerpt"
+							numberOfLines={ 3 }
 						>
-							{ record.link }
-						</ExternalLink>
-					) : null }
-					{ record
-						? decodeEntities( record?.description?.rendered )
-						: null }
+							{ stripHTML( record.excerpt.rendered ) }
+						</Truncate>
+					) }
+					<SidebarNavigationSubtitle>
+						Details
+					</SidebarNavigationSubtitle>
+					<SidebarDetails
+						details={ getPageDetails( {
```

Comment:
This function is a bit weird, Can't we instead transform it to a `<PageDetails id={pageId}  />` instead and collocate all the logic?